### PR TITLE
Setup script for getting started

### PR DESCRIPTION
**Pull request by Yukaii**
**Notice: Before fast-foward to this commit on your local machine, backup config files to other place, otherwise your local config files will be deleted.**

Ignored config files are:

```
config.json
public/js/common.js
.sequelizerc
```

#### Changes

* Ignore config files in git repository, and instead create an example config copy, keeping git workspace clean and easier to commit
* Write a setup script(`bin/setup`) to create config file from example copy, and do package installing stuff
* Update corresponding part in README

### CODE REVIEW

File: bin/setup [hunk 1]
```diff
@@ -0,0 +1,49 @@
+#!/bin/bash
+
+# run command at repo root
+CURRENT_PATH=$PWD
+cd $(git rev-parse --show-toplevel)
+
+if ! type npm > /dev/null
+then
+  echo "npm is not installed, please install via nvm(recommended) or package manager"
```
Comment (jackycute) [inline bin/setup]: Please don't recommend anything right here.

File: README.md [hunk 2]
```diff
@@ -43,7 +43,7 @@ Get started
 ---
 
 1. Download a release and unzip or clone into a directory
-2. Enter the directory and type `npm install && bower install`, which will install all the dependencies
+2. Enter the directory and type `bin/setup`, which will install all the dependencies and create config file
```
Comment (jackycute) [inline README.md]: Please add more description for prerequisite of npm and need a bash to run the setup script.

File: bin/setup (hunk 1 above)
Comment (Yukaii) [inline bin/setup]: Removed, and add Node.js official website link

File: README.md (hunk 2 above)
Comment (Yukaii) [inline README.md]: I change this line to:

> Enter the directory and type `bin/setup`, which will install npm/bower dependencies and create configs. The setup script is written in Bash, you would need bash as a prerequisite.

Is this OK?

Comment (jackycute) [inline README.md]: OK

**Comment by jackycute** [comment]
Very well, let's merge it now.

**Pull request merged**